clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy

```diff
@@ -160,9 +160,35 @@ class ServerGroupController {
   @PostFilter("hasPermission(filterObject?.application, 'APPLICATION', 'READ')")
   @PostAuthorize("@authorizationSupport.filterForAccounts(returnObject)")
   @RequestMapping(value = "/serverGroups", method = RequestMethod.GET)
-  List getServerGroupsByApplications(@RequestParam(value = 'applications') List<String> applications,
-                                     @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
-    applications.collectMany { summaryList(it, cloudProvider) }
+  List getServerGroups(@RequestParam(required = false, value = 'applications') List<String> applications,
+                       @RequestParam(required = false, value = 'serverGroupNames') List<String> serverGroupNames,
+                       @RequestParam(required = false, value = 'cloudProvider') String cloudProvider) {
+    if ((applications && serverGroupNames) || (!applications && !serverGroupNames)) {
+      throw new IllegalArgumentException("Provide either 'applications' or 'serverGroupNames' parameter, but not both");
+    }
+
+    if (applications) {
+      return getServerGroupsForApplications(applications, cloudProvider)
+    } else {
+      return getServerGroupsForNames(serverGroupNames)
+    }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForApplications(List<String> applications, String cloudProvider) {
+    return applications.collectMany { summaryList(it, cloudProvider) }
+  }
+
+  private List<ServerGroupViewModel> getServerGroupsForNames(List<String> serverGroupNames) {
+    return serverGroupNames.collect {
+      def segments = it.split(':')
+      if (segments.size() != 3) {
+        throw new IllegalArgumentException("Expected serverGroupNames in the format [account1:region1:name1,account2:region2:name2] but got " + serverGroupNames)
+      }
+
+      def (account, region, name) = segments;
+      def serverGroup = getServerGroup(account, region, name)
```

Comment:
If a single server group is not found, the entire call fails with a 404.  Does this seem appropriate, or should the results be returned without missing data?

Reply:
I don't think I would expect this to throw a 404 ... more generally I wouldn't expect any root endpoint (ie. `/serverGroups`) that returns a collection and takes filter query parameters to 404. 

There's a whole other discussion / argument to be had about our inconsistent APIs and the pain associated with _only_ returning a collection vs. an envelope. The latter would allow us to return both the collection of matched server groups and a set of missing server groups if desired.

Reply:
Not sure how many server groups are likely to be in this request ... might be an inefficient lookup if there are _lots_.

Reply:
>  I wouldn't expect any root endpoint (ie. /serverGroups) that returns a collection and takes filter query parameters to 404

Sure.  I was thinking of this less as a filter query (tell me what's there that satisfies this query) and more as a batched "get by id" (I know these things exist, give me them).  For my use case, the former is actually more convenient.  I will address this.

> might be an inefficient lookup if there are lots

For search v2, this will be small batches (25 right now).  Is there a more efficient mechanism I could be using?

Reply:
Let's see how it goes ... I'm not sure ... especially if the server groups are somewhat spread across region / application.